src/test/java/com/microsoft/sqlserver/jdbc/fedauth/PooledConnectionTest.java

```diff
@@ -435,27 +436,15 @@ public void testDSPooledConnectionAccessTokenCallbackClassExceptions() throws Ex
 
         // User/password is not required for access token callback
         AbstractTest.updateDataSource(accessTokenCallbackConnectionString, ds);
+
         ds.setAccessTokenCallbackClass(AccessTokenCallbackClass.class.getName());
         ds.setUser("user");
-        SQLServerPooledConnection pc;
-
-        // Should fail with user set
-        try {
-            pc = (SQLServerPooledConnection) ds.getPooledConnection();
-            fail(TestResource.getResource("R_expectedFailPassed"));
-        } catch (SQLServerException e) {
-            assertTrue(e.getMessage().matches(TestUtils.formatErrorMsg("R_AccessTokenCallbackWithUserPassword")));
-        }
-
-        ds.setUser("");
         ds.setPassword(UUID.randomUUID().toString());
+        SQLServerPooledConnection pc;
 
-        // Should fail with password set
-        try {
-            pc = (SQLServerPooledConnection) ds.getPooledConnection();
-            fail(TestResource.getResource("R_expectedFailPassed"));
-        } catch (SQLServerException e) {
-            assertTrue(e.getMessage().matches(TestUtils.formatErrorMsg("R_AccessTokenCallbackWithUserPassword")));
+        pc = (SQLServerPooledConnection) ds.getPooledConnection();
+        try (Connection conn1 = pc.getConnection()) {
+            assertNotNull(conn1);

```

Comment:
this isn't necessary can just be removed. getConnection will always return a connection object or throw an exception. Since this test case isn't valid anymore it can just be removed

Reply:
As per the review comment at https://github.com/microsoft/mssql-jdbc/pull/2549#discussion_r1861036997, I have added a test case to detect a successful connection once the username and password are set.